pkg/kube/kube_handler.go

```diff
@@ -649,36 +649,49 @@ func filterAllowedNamespaces(userClientset combinedClientsetInterface, namespace
 			allowedNamespaces = append(allowedNamespaces, namespace)
 		}
 	}
-	namespaces.Items = allowedNamespaces
-	return namespaces, nil
+	return allowedNamespaces, nil
+}
+
+func filterActiveNamespaces(namespaces []corev1.Namespace) []corev1.Namespace {
+	readyNamespaces := []corev1.Namespace{}
+	for _, namespace := range namespaces {
+		if namespace.Status.Phase == corev1.NamespaceActive {
+			readyNamespaces = append(readyNamespaces, namespace)
+		}
+	}
+	return readyNamespaces
 }
 
 // GetNamespaces return the list of namespaces that the user has permission to access
 func (a *userHandler) GetNamespaces() ([]corev1.Namespace, error) {
 	// Try to list namespaces with the user token, for backward compatibility
 	namespaces, err := a.clientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		// TODO: #1763 Check if we have a service token for getting namespaces on other clusters.
 		if k8sErrors.IsForbidden(err) {
 			// The user doesn't have permissions to list namespaces, use the current serviceaccount
 			namespaces, err = a.svcClientset.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
 			if err != nil && k8sErrors.IsForbidden(err) {
 				// If the configured svcclient doesn't have permission, just return an empty list.
 				return []corev1.Namespace{}, nil
 			}
-
-			// Only if we obtained the namespaces from the svc client do we filter it using
-			// the user clientset.
-			namespaces, err = filterAllowedNamespaces(a.clientset, namespaces)
-			if err != nil {
-				return nil, err
-			}
 		} else {
 			return nil, err
 		}
 	}
 
-	return namespaces.Items, nil
+	// Filter namespaces in which the user has permissions to write (secrets) only
+	namespaceList, err := filterAllowedNamespaces(a.clientset, namespaces)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Not sure why this filter has been moved out to here: isn't it a superfluous (and relatively expensive) check if the request was done with the user's own credentials? (if you decide to move it back, just remember to update the wording of the second test which changed accordingly, I think)

Reply:
It's not superfluous since it's covering the use case in which a user has permissions to list namespaces (which is harmless) but don't have permissions to access resources in all the namespaces (not an admin). This prevents people from using namespaces that they don't have access.

I'm thinking on the case, for example, I (andres@corp.com) have permissions to list namespaces but only have write access to my namespace `foo`. If `bar` exists, Kubeapps will autoselect the first namespace in alphabetical order (which is `bar`) so by default I will get a forbidden error if I try to do anything without changing the namespace.